### ISSUE TITLE
Unexpected Response :Uploading a cover image(>30MB or unknown extension).

### ISSUE THREAD

**Issue by Aj-KamaL** [internetarchive/openlibrary#3208]
The "Add Cover Image" input form "Choose File" that takes an input image(File)  form the user does not handle corner cases, 
1. [uploading an image>30 MB](https://github.com/internetarchive/openlibrary/issues/2390#issue-493620943),
2. uploading a non-image file(other than JPG, GIF or PNG).

### Evidence / Screenshot (if possible)
Following error page opens up in 1 and 2 cases respectively:
![image](https://user-images.githubusercontent.com/23471306/76924176-8ecb7580-68fb-11ea-95e5-2edd19d72536.png)

### Steps to Reproduce

1. Click on "Manage Covers" inside the cover page of any borrowed book. 
2. Click on "Choose File" to upload an image.
3. Select a non-image file/ or file with [size> 30MB].(https://github.com/internetarchive/openlibrary/issues/2390#issue-493620943).
4. Click Submit.

* Actual: 
Throw an unexpected Response.  
* Expected: 
**Must not allow the user to proceed to click "Submit" with an invalid image.**
### Details

- **Logged in (Y/N)?** Y 
- **Browser type/version?** 
- **Operating system?** 
- **Environment (prod/dev/local)?** prod

### Proposal & Constraints

Check the file extension and size. 
@Pop an alert and prevent the user from even uploading invalid images.

**Comment by tabshaikh** [comment]
GIF book cover's should that too be allowed? Also are GIF book covers present for books on ol?

**Comment by seabelis** [comment]
Good question. I've never tried it. Surely we don't want animated gifs nor is there a need to support transparency. I don't check file types of covers so I can't really say if they are present or not.

**Comment by BrittanyBunk** [comment]
@tabshaikh I agree with @seabelis. I would say it's ok for there to be GIF's, just no animated ones. I uploaded one and then took it down. I see no reason why they weren't. I upload png's, but I take from websites and it's quite likely that those are in GIF. 

Outside of that, I already wrote about this in #2390.

I just don't see why anyone would upload anything that's not an image file to a cover. I'm not even sure why this issue is an issue?

**Comment by Aj-KamaL** [comment]
I agree with @BrittanyBunk and @seabelis. GIF is a very common file extension and even if the user attempts to upload an animation, the GIF changes to an image.
About upload transparency, ["Unrestricted File Upload forms are a Major Security Threat"](https://owasp.org/www-community/vulnerabilities/Unrestricted_File_Upload).
**Whitelisting File Content/type** and **Restricting File Size** are very common and very basic protections. 
Will it be better if this issue is categorised as a feature, than a bug?

**Comment by BrittanyBunk** [comment]
@Aj-KamaL I just think this should be closed, because I already addressed this issue in the first part and you did for the 2nd part. So it's pretty finished from what I see.

**Comment by tfmorris** [comment]
The PR addressing this hasn't been merged, so this shouldn't be closed yet.

**Comment by jimchamp** [comment]
#9966 adds a number of server-side image validations, which, I believe, closes this issue.  Feel free to reopen this if I'm misunderstanding something.
